src/components/blipDatepicker/index.js

```diff
@@ -0,0 +1,615 @@
+    this._renderMonth()
+    this._renderTime()
+  }
+
+  set selectedDay(newDate) {
+    this._selectedDay = new Date(newDate)
+    this._selectedPeriod = undefined
+    this._clearSelection()
+    this._renderSelection(newDate)
+  }
+
+  set selectedPeriod(newPeriod) {
+    this._selectedPeriod = new Period(newPeriod.startDate, newPeriod.endDate)
+    this._selectedDay = undefined
+    this._renderSelection()
+  }
+
+  set showPrev(visibility) {
+    this._setElementVisibility(this._prevButton, visibility)
+  }
+
+  set showNext(visibility) {
+    this._setElementVisibility(this._nextButton, visibility)
+  }
+
+  createElement(monthDate) {
+    this._datepicker = this._createHTMLElement('div', BlipDatepicker.style.datepicker)
+
+    this._createMonthTable()
+    this._createDateSelector()
+    if (this.hasTime) {
+      this._createTimeStructure()
+    }
+
+    this._addEventListeners()
+
+    this.monthDate = new Date(monthDate)
+
+    return this._datepicker
+  }
+
+  _createMonthTable() {
+    var monthTable = this._createHTMLElement('table', BlipDatepicker.style.monthTable)
+    var tableHead = monthTable.createTHead()
+    var tableBody = monthTable.createTBody()
+
+    var headerRow = tableHead.insertRow()
+
+    var titleCell = this._createHTMLElement('th')
+    var prevCell = this._createHTMLElement('th')
+    var nextCell = this._createHTMLElement('th')
+
+    var monthTitle = this._createHTMLElement('span', BlipDatepicker.style.monthTitle)
+    var yearInput = this._createHTMLElement('input', BlipDatepicker.style.yearInput)
+    yearInput.type = 'number'
+
+    var createButton = (icon, value, ...classes) => {
+      var button = html`<button class="${BlipDatepicker.style.monthButton}">${raw(icon)}</button>`
+      if (classes) classes.forEach(className => button.classList.add(className))
+      button.value = value
+      return button
+    }
+
+    var prevButton = createButton(ArrowLeft, '-1', BlipDatepicker.style.monthPrev)
+    var nextButton = createButton(ArrowRight, '1', BlipDatepicker.style.monthNext)
+
+    titleCell.colSpan = '5'
+    titleCell.appendChild(monthTitle)
+    titleCell.appendChild(yearInput)
+    prevCell.appendChild(prevButton)
+    nextCell.appendChild(nextButton)
+
+    headerRow.appendChild(prevCell)
+    headerRow.appendChild(titleCell)
+    headerRow.appendChild(nextCell)
+
+    var weekdayRow = tableBody.insertRow()
+    weekdayRow.classList.add(BlipDatepicker.style.weekdayWeek)
+    for (var day = 0; day < BlipDatepicker.weekSize; day++) {
+      var weekday = weekdayRow.insertCell()
+      weekday.classList.add(BlipDatepicker.style.weekDay)
+      weekday.innerText = this.i18n.weekdays[day][0]
+      weekday.title = this.i18n.weekdays[day]
+    }
+
+    var daysRows = []
+    var monthDays = []
+    for (var week = 0; week < BlipDatepicker.monthRows; week++) {
+      var weekRow = tableBody.insertRow()
+      for (day = 0; day < BlipDatepicker.weekSize; day++) {
```

Comment:
This `day` variable scope is dangerous, because if we have some variable with `day` name in other place of this function scope, then we'll drop in the "JavaScript hoisting" problem (https://developer.mozilla.org/pt-BR/docs/Glossario/Hoisting). Try to redeclare it using `let` or avoid `for` iteration using some functional way.